### PR TITLE
fix: temporary fix for cross-modal-search index flow to work on Jina 2.0

### PR DESCRIPTION
**Pull request by winstonww**
The fix temporarily reverts previous change that ports CLIPEncoder to Jinahub because a hang occurs. Also remove the lines `f.use_rest_gatway()` as the function is now removed.

### CODE REVIEW

File: cross-modal-search/app.py [hunk 1]
```diff
@@ -23,8 +23,7 @@ def config():
 
 
 def index_restful():
-    flow = Flow().load_config('flows/flow-index.yml')
-    flow.use_rest_gateway()
+    flow = Flow(protocol='http').load_config('flows/flow-index.yml', override_with={'protocol': 'http'})
```
Comment (numb3r3) [inline cross-modal-search/app.py:26]: To keep the code simple

```suggestion
    flow = Flow().load_config('flows/flow-index.yml', override_with={'protocol': 'http'})
```
I'm now sure whether my revision can work, please test it.

Comment (JoanFM) [inline cross-modal-search/app.py:26]: `override_with` protocol should be removed as it is not affecting what u think

Comment (JoanFM) [reply cross-modal-search/app.py:26]: the `protocol` should be in the `flow-index.yml`

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: cross-modal-search/flows/flow-index.yml [hunk 2]
```diff
@@ -6,13 +6,15 @@ with:
...
+    # uses: 'jinahub+docker://ImageNormalizer'
     uses: pods/image-load.yml
     shards: $JINA_PARALLEL
     read_only: true
     needs: [gateway]
   - name: image_encoder                               # encode images into embeddings with CLIP model
-    uses: 'jinahub+docker://ClipImageEncoder'
-    volumes: $HOME/.cache/clip:/root/.cache/clip
+    # uses: 'jinahub+docker://CLIPImageEncoder'
+    # volumes: $HOME/.cache/clip:/root/.cache/clip
```
Comment (numb3r3) [inline cross-modal-search/flows/flow-index.yml:16]: remove the unused codes

File: cross-modal-search/flows/flow-query.yml [hunk 3]
```diff
@@ -17,8 +18,9 @@ pods:
 #    needs: loader
   - name: image_encoder                   # encode query image into embeddings with CLIP model
     polling: any
-    uses: 'jinahub+docker://ClipImageEncoder'
-    volumes: $HOME/.cache/clip:/root/.cache/clip
+     # uses: 'jinahub+docker://ClIPImageEncoder'
+     # volumes: $HOME/.cache/clip:/root/.cache/clip
```
Comment (numb3r3) [inline cross-modal-search/flows/flow-query.yml:22]: same

File: cross-modal-search/requirements.txt [hunk 4]
```diff
@@ -1,7 +1,7 @@
-jina[hub, http]==2.0.0rc5
+jina[hub, http]==2.0.5
```
Comment (numb3r3) [inline cross-modal-search/requirements.txt:1]:
```suggestion
jina[hub, http]~=2.0.0
```

**Review by numb3r3 — CHANGES_REQUESTED** [review]

File: cross-modal-search/app.py (hunk 1 above)
Comment (winstonww) [reply cross-modal-search/app.py]: As the `index` method which uses GRPC protocol also load its config from `flow-index.yml`, in this case I think protocol may be better defined in `app.py`? The override_with keyword seems to work in my environment:
```
[10:03:57] (env) winstonwong@Winstons-MacBook-Pro ~/jina/examples/examples/cross-modal-search $ python3 app.py -t index_restful
         loader@56734[L]:ready and listening
         image_normalizer@56734[L]:ready and listening
         image_encoder@56734[L]:ready and listening
         image_vector_indexer@56734[L]:ready and listening
         image_kv_indexer@56734[L]:ready and listening
         text_encoder@56734[L]:ready and listening
         text_indexer@56734[L]:ready and listening
         join_all@56734[L]:ready and listening
         gateway@56734[L]:ready and listening
         Flow@56734[I]:🎉 Flow is ready to use!
        🔗 Protocol:          HTTP
        🏠 Local access:      0.0.0.0:45678
        🔒 Private network:   172.16.10.4:45678
        💬 Swagger UI:                http://localhost:45678/docs
        📚 Redoc:             http://localhost:45678/redoc
```
Thank you

Comment (winstonww) [reply cross-modal-search/app.py]: Yep, I believe the proposed change works for me. Please see [e11b980](https://github.com/jina-ai/examples/pull/672/commits/e11b980488645a32eac1fbe3a7c9ca27def8d850) for the change.

File: cross-modal-search/flows/flow-index.yml (hunk 2 above)
Comment (winstonww) [reply cross-modal-search/flows/flow-index.yml]: Addressed, please see [e11b980](https://github.com/jina-ai/examples/pull/672/commits/e11b980488645a32eac1fbe3a7c9ca27def8d850) for the change.

File: cross-modal-search/flows/flow-query.yml (hunk 3 above)
Comment (winstonww) [reply cross-modal-search/flows/flow-query.yml]: Addressed, please see [e11b980](https://github.com/jina-ai/examples/pull/672/commits/e11b980488645a32eac1fbe3a7c9ca27def8d850) for the change.

File: cross-modal-search/requirements.txt (hunk 4 above)
Comment (winstonww) [reply cross-modal-search/requirements.txt]: Addressed, please see [e11b980](https://github.com/jina-ai/examples/pull/672/commits/e11b980488645a32eac1fbe3a7c9ca27def8d850) for the change. Thank you.

**Review by numb3r3 — APPROVED** [review]

File: cross-modal-search/app.py [hunk 5]
```diff
@@ -23,8 +23,7 @@ def config():
 
 
 def index_restful():
-    flow = Flow().load_config('flows/flow-index.yml')
-    flow.use_rest_gateway()
+    flow = Flow().load_config('flows/flow-index.yml', override_with={'protocol': 'http'})
```
Comment (JoanFM) [inline cross-modal-search/app.py:26]: this I am not sure how is it working? The `protocol` should be in the `flow-index.yml` as an ENV var

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: cross-modal-search/app.py (hunk 5 above)
Comment (winstonww) [reply cross-modal-search/app.py:26]: As discussed, a test is added to core to test`Flow.load_config` with `override_with` [here](https://github.com/jina-ai/jina/pull/2901).

**Review by JoanFM — APPROVED** [review]

**Pull request merged**